lib/matplotlib/backends/_backend_tk.py

```diff
@@ -649,12 +649,59 @@ def _set_image_for_button(self, button):
         path_large = path_regular.with_name(
             path_regular.name.replace('.png', '_large.png'))
         size = button.winfo_pixels('18p')
+
+        # Nested functions because ToolbarTk calls  _Button.
+        def _get_color(color_name):
+            # `winfo_rgb` returns an (r, g, b) tuple in the range 0-65535
+            return button.winfo_rgb(button.cget(color_name))
+
+        def _is_dark(color):
+            if isinstance(color, str):
+                color = _get_color(color)
+            return max(color) < 65535 / 2
+
+        def _recolor_icon(image, color):
+            image_data = np.asarray(image).copy()
+            black_mask = (image_data[..., :3] == 0).all(axis=-1)
+            image_data[black_mask, :3] = color
+            return Image.fromarray(image_data, mode="RGBA")
+
         # Use the high-resolution (48x48 px) icon if it exists and is needed
         with Image.open(path_large if (size > 24 and path_large.exists())
                         else path_regular) as im:
             image = ImageTk.PhotoImage(im.resize((size, size)), master=self)
-        button.configure(image=image, height='18p', width='18p')
-        button._ntimage = image  # Prevent garbage collection.
+            button._ntimage = image
+
+            # create a version of the icon with the button's text color
+            foreground = (255 / 65535) * np.array(
+                button.winfo_rgb(button.cget("foreground")))
+            im_alt = _recolor_icon(im, foreground)
+            image_alt = ImageTk.PhotoImage(
+                im_alt.resize((size, size)), master=self)
+            button._ntimage_alt = image_alt
+
+        if _is_dark("background"):
+            button.configure(image=image_alt)
+        else:
+            button.configure(image=image)
+        if (
+            isinstance(button, tk.Checkbutton)
+            and button.cget("selectcolor") != ""
+        ):
+            if self._windowingsystem != "x11":
+                selectcolor = "selectcolor"
+            else:
+                # On X11, selectcolor isn't used directly for indicator-less
+                # buttons.
+                r1, g1, b1 = _get_color("selectcolor")
+                r2, g2, b2 = _get_color("activebackground")
+                selectcolor = ((r1+r2)/2, (g1+g2)/2, (b1+b2)/2)
+            if _is_dark(selectcolor):
+                button.configure(selectimage=image_alt)
+            else:
+                button.configure(selectimage=image)
```

Comment:
This section could use a comment along the lines

> Checkbutton switches the background color ("selectcolor") depending on
> check state. Adjust the button image accordingly to make sure it has
> enough contrast to the background.
> make

Reply:
Have updated